index-scheduler/Cargo.toml

```diff
@@ -19,6 +19,7 @@ dump = { path = "../dump" }
 enum-iterator = "1.1.3"
 file-store = { path = "../file-store" }
 log = "0.4.14"
+meilisearch-auth = { path = "../meilisearch-auth" }
```

Comment:
What do you think about that @Kerollmops?

I think we were against it some time ago, but I can't remember why.

Reply:
I'm not @Kerollmops, but if I had to review this code, I guess my objection would be around *coupling*.

Adding the dependency is a symptom that the `index-scheduler` depends on the implementation of the `meilisearch-auth`, which might not be what we want.

The various approaches I can suggest:

1. Move the `AuthFilter` to `meilisearch-types`. I don't feel like this approach actually reduces the coupling between `meilisearch-auth` and the `index-scheduler`, IMO it just misplaces some of the `AuthFilter` implementation in the `meilisearch-types` crate.
2. Create an `AuthFilter` trait in `meilisearch-types`, implement it on `meilisearch-auth::AuthFilter`, then accept any type implementing the trait in the `index-scheduler`. This actually reduces the coupling (for instance it becomes possible to implement a `MockAuthFilter` in the `index-scheduler` for tests and we no longer need `AuthFilter::with_allowed_indexes`). It increases the complexity, though, and I don't really like adding a trait that will have a single production implementation.
3. Pass a single optional list of `allowed_indexes` like we did before. In the previous implementation, the list is however incorrect when using tenant token, which happened not to be the case, but it looks to me like a footgun to expose it as part of the `AuthFilter` interface. It is unclear to me that we could somehow merge the two lists of allowed indexes between the API key and the tenant token is a way that always works.
4. Pass a single `Option<(Vec<UuidPattern>, Vec<UuidPattern>)>` and have the the caller make sure to implemented the logic to extract the equivalent of `is_authorized_index` from this object. IMO this reduces coupling at the price of encapsulation, because the caller is expected to know how to correctly use a complicated type so as to keep its invariants, rather than the type being named and keeping its invariant on its own.
5. Wrap the type in (4) in a struct that exposes the `is_authorized_index` method. Now we have the same issue of where to put this struct, but for this struct instead of `AuthFilter`. Also, it is unclear how this struct is significantly different from `AuthFilter` itself.
6. Accept the coupling and move on by doing what the current PR does. It was my choice but I can revisit it.

The solutions above rely on the conceptual argument of unnecessary coupling, but maybe there are other arguments against this added dependency. Feel free to discuss I'm ready to implement most of the solutions above if needed.

Reply:
That's maybe not a bad idea anymore. I think the reason was that we didn't want cross-dependencies. I am not sure about the drawbacks.